lib/offsite_payments/integrations/payu_in.rb

```diff
@@ -117,7 +117,12 @@ def invoice_ok?( order_id )
 
         # Order amount should be equal to gross - discount
         def amount_ok?( order_amount, order_discount = BigDecimal.new( '0.0' ) )
-          BigDecimal.new( original_gross ) == order_amount && BigDecimal.new( discount.to_s ) == order_discount
+          bigdecimal_discount = begin
+            BigDecimal.new( discount.to_s )
+          rescue ArgumentError => e
```

Comment:
This seems a bit heavy handed if `discount.to_d` has the same zero coercion behaviour we're looking for?

Ruby 2.3.3:
```
irb(main):002:0> BigDecimal.new("0.")
=> #<BigDecimal:7f80dc905ab0,'0.0',9(18)>
irb(main):003:0> "0.".to_d
=> #<BigDecimal:7f80dc90f4c0,'0.0',9(18)>
```

Ruby 2.4.2 w/ BigDecimal 1.3.2:
```
irb(main):004:0> "0.".to_d
=> 0.0
irb(main):005:0> BigDecimal.new("0.")
ArgumentError: invalid value for BigDecimal(): "0."
```

Reply:
Oh wow, this will make things a whole lot simpler. In that case, instead of `discount.to_s`, using `discount.to_d` should do the job. Thanks so much!